Doc & README rework for new releases system (#2171)
RogueException/Discord.Net: docs/guides/getting_started/terminology.md

Discord.Net is split into a core library and two different
implementations - `Discord.Net.Core`, `Discord.Net.Rest`, and
`Discord.Net.WebSocket`.

> As a common bot developer, you will only need to use `Discord.Net.WebSocket`,

Reviewer: What do you mean by a "common bot developer" as opposed to just a "bot developer"?

Author: Developers who implement the REST interactions system [do not use websocket](https://github.com/Rozen4334/Barriot/blob/master/Barriot.Core/Program.cs#L25) at all. Because it is so uncommon for people to use this system in dnet, I chose to mention that you commonly do not need it.

Reviewer: Alright, that makes sense. Perhaps this sounds better to remove the need to clarify `common`?
```suggestion
You will typically only need to use `Discord.Net.WebSocket`,
```
